### DISCUSSION TITLE
关于 resources_custom 存在的必要性讨论

### DISCUSSION THREAD

**Discussion by Xm798** [Arondight/Adachi-BOT#610] [General]
目前我们已经实现了所有配置文件的自托管，由于`overview`样式的改动，导致原作者的 json 已经不适用于本项目，那么`resources_custom`是否还有存在的必要性呢

**Comment by mark9804** [comment]
嗯……我觉得存在的必要性在逐渐趋近于 0 但还没到 0，但是可以把一些目录（例如我们自己的 `json` 文件）给脱离出去，但是比如说 [人物头像](https://github.com/Arondight/Adachi-BOT/tree/master/resources/Version2/thumb/character) 还是可以继续用的，这些都是标准头像，谁做都一样

**Comment by Arondight** [comment]
我想保留，可以继续用原作者的，无非是数组里面如果只有四个元素，那我在特定两个位置插俩空串嘛，只不过前端拿到空串需要处理一下，原作者还是很猛的，比我要猛地多，这些资源文件应该全都是手撸的，图也是挨个 ps 的

**Comment by Xm798** [comment]
忘记还有图片了，图片的话确实还可以保留，json 的话还保留在 custom 里感觉略有些麻烦

**Reply by Arondight** [reply]
我把 json 的更新给掐了，只加新文件不做更新就可以了

**Comment by mark9804** [comment]
> 那我在特定两个位置插俩空串嘛，只不过前端拿到空串需要处理一下

阿哲，刚提完更新才看到这个 commit，不过现在的前端倒也不会炸，就是难看了点，我再提一个 commit

**Reply by Arondight** [reply]
我还没有做……只是计划

**Reply by Arondight** [reply]
不过想想似乎写个程序找命座信息不全的 json 给插俩空串也没啥意义，本来更新脚本就是 dev 分支，到头来遇到空串还得手动写

**Reply by Arondight** [reply]
似乎直接在前端做这个最好，json 就不改了，直接把数据结构改掉，我开完了会提交一个

**Comment by mark9804** [comment]
> 原作者还是很猛的，比我要猛地多，这些资源文件应该全都是手撸的，图也是挨个 ps 的

怎么说呢，就，精神可嘉，但是单评价前端的话，如果他这套 css 继续用下去，技术债最终会越积越多的，想加任何一个新元素都要做大量的重写和适配（手动修改图片尺寸也算适配），老大哥当时[自己拖海染套圣遗物图片](https://github.com/Arondight/Adachi-BOT/commit/780d022a503e7099d4f271ae8a2879a62668fd53)的时候也见识过这个威力了

很多东西如果当时写基础结构的时候多痛苦一两个小时的话，后面可以少痛苦一两百个小时，我觉得从这个角度来说 silvery 的“猛”有点可惜

> 似乎直接在前端做这个最好，json 就不改了，直接把数据结构改掉

倒也不一定要改，我想到魔改的操作了，不过要是能把传进来的数据结构改成下面这样也行，反正我在前端也打算处理成下面这样的

```json
[
    {"一": "阿巴阿巴"},
    {"二": "阿巴阿巴"},
    {"三（如果有）": "阿巴阿巴"},
    {"四": "阿巴阿巴"},
    {"五（如果有）": "阿巴阿巴"},
    {"六": "阿巴阿巴"}
]
```

**Reply by Arondight** [reply]
我准备在前端和bot中凡是遇到四个元素的命座数组豆插俩空串上去

**Reply by mark9804** [reply]
不存在就让他不存在吧，我是打算把空串当作不存在的，如果有四个不为空的命座信息，我就显示四栏；如果有六个不为空的命座信息，我就显示六栏

**Reply by Arondight** [reply]
那这样一搞原作者的json可以继续用了，我感觉原作者这个生猛度，能够拿来用是最好的，原地一躺直接开摆

**Reply by Arondight** [reply]
连带前端一起改了，现在原作者的 json 三五命显示是个空，至少不会出错了
